.github/workflows/tests.yml

```diff
@@ -413,3 +413,24 @@ jobs:
         with:
           name: ember-exam-execution-${{ matrix.browser }}-${{ hashFiles('./app/assets/javascripts/discourse/test-execution-*.json') }}
           path: ./app/assets/javascripts/discourse/test-execution-*.json
+
+   merge:
+    if: github.repository == 'discourse/discourse' && github.ref == 'refs/heads/main'
+    runs-on: debian-12
+    needs: build
+    steps:
+      - name: Merge Artifacts
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: failed-system-test-screenshots
+          pattern: failed-system-test-screenshots-*
+          delete-merged: true
+          separate-directories:: false
+
+      - name: Merge Artifacts
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: flaky-test-reports
+          pattern: flaky-test-reports-*
+          delete-merged: true
+          separate-directories: false
```

Comment:
@davidtaylorhq @CvX I went with merging the artifacts instead following the [migration guide](https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md#multiple-uploads-to-the-same-named-artifact). I looked into updating our internal plugin that relies on these artifacts but the change is much more complicated since I now have to pattern match on all artifacts and download each artifact individually.